webapp/src/components/StartingMaterialInformation.vue

```diff
@@ -1,73 +1,70 @@
+        <label for="name">Name</label>
+        <StyledInput id="name" v-model="Name" :readonly="!isEditable" />
       </div>
     </div>
     <div class="form-row">
-      <div class="form-group col-md-3">
-        <label id="collections" class="mr-2">Collections</label>
-        <div>
-          <CollectionList aria-labelledby="collections" :collections="Collections" />
-        </div>
-      </div>
-      <div class="form-group col-md-3">
-        <label for="chemform" class="mr-2">Chemical formula</label>
-        <span class="form-control-plaintext" readonly="true">
-          <ChemicalFormula :formula="item.chemform" />
+      <div class="form-group col-lg-3 col-sm-4">
+        <label for="chemform">Chemical formula</label>
+        <ChemFormulaInput v-if="isEditable" id="chemform" v-model="ChemForm" />
+        <span v-if="!isEditable" class="form-control-plaintext" readonly>
+          <ChemicalFormula id="chemform" :formula="ChemForm" />
         </span>
       </div>
-      <div class="form-group col-md-3">
-        <label for="supplier" class="mr-2">Supplier</label>
-        <input
-          id="supplier"
-          :value="item.supplier"
-          class="form-control-plaintext"
-          readonly="true"
-        />
+      <div class="form-group col-lg-3 col-sm-4">
+        <label for="supplier">Supplier</label>
+        <StyledInput id="supplier" v-model="Supplier" :readonly="!isEditable" />
+      </div>
+      <div class="form-group col-lg-3 col-sm-4">
+        <label for="purity">Chemical purity</label>
+        <StyledInput id="purity" v-model="ChemicalPurity" :readonly="!isEditable" />
       </div>
-      <div class="form-group col-md-3">
-        <label for="purity" class="mr-2">Chemical purity</label>
-        <input
-          id="purity"
-          :value="item.chemical_purity"
-          class="form-control-plaintext"
-          readonly="true"
+    </div>
+    <div class="form-row">
+      <div class="form-group col-lg-3 col-sm-4">
+        <label for="date-acquired">Date acquired</label>
+        <StyledInput
+          id="date-acquired"
+          type="date"
+          v-model="DateAcquired"
+          :readonly="!isEditable"
         />
       </div>
+      <div class="form-group col-lg-3 col-sm-4">
+        <label for="date-opened">Date opened</label>
+        <StyledInput id="date-opened" type="date" v-model="DateOpened" :readonly="!isEditable" />
+      </div>
+      <div class="form-group col-lg-3 col-sm-4">
+        <label for="location">Location</label>
+        <StyledInput id="location" v-model="Location" :readonly="!isEditable" />
+      </div>
     </div>
 
-    <label for="location" class="mr-2">Location</label>
-    <input id="location" :value="item.location" class="form-control-plaintext" readonly="true" />
+    <div class="form-row">
+      <div class="form-group col-lg-3 col-sm-4">
+        <label for="cas">CAS</label>
+        <StyledInput id="cas" v-model="CAS" :readonly="!isEditable" />
```

Comment:
Not for this PR but I'd like CAS to be something similar to the toggleable form; its a string field that will rarely have to change and it should do something "special" when locked in, e.g., link to a CAS registry search for that item

Reply:
agreed. I would also like to have a way of automatically populating many of these fields based on CAS or another code, but will leave that for the future.

Reply:
issue #658